src/amp/components/Experiment.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+
+export interface ExperimentModel {
```

Comment:
I think the types and functions should be moved to src/amp/lib, in order to keep amp/components/ more strictly related to the rendering.

Reply:
Happy to do this, but worth noting that I took this format from the existing Analytics component and module. If we move this around for the `Experiment` component, we might want to do the same for the `Analytics` component and any others that match this.

Reply:
Would be good to get more thoughts on this. I personally think that types should be as far as possible declared at the *point of use*. So here would make sense in this case.

Reply:
I agree with Nic, if a type is only used in one place then it makes sense to declare it right there. It's easier for others to reason about / maintain the code when it's not distributed

Reply:
I'd also +1 Jerome's other point around keeping the component related to rendering. We don't have any strict rules around this but I find things easier to manage when files do one thing and have one export. Sometimes it feels wrong to break something out into a new file though and keeping it local is the way to go. If there had to be a dogmatic rule then it would be to 'always consider maintainability'

In DCR (web) we have a lib folder with a series of files in it, one file per function. It sounds weird but it makes finding what you need easier and prevents the `utils.js` explosion that normally happens on most projects. Not sure if that helps 🤷

Reply:
I've moved non-component code into the `src/lib` directory to keep the code responsible for rendering separate and cleaner.